pyproject.toml

```diff
@@ -8,57 +8,54 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = ">3.9.7, <3.12"
-openai = ">=1.0.0"
-python-dotenv = ">=0.2.0"
-tiktoken = ">=0.4.0"
-importlib-metadata = ">=6.8.0"
-tokenizers = "*"
-click = "*"
-jinja2 = "^3.1.2"
-certifi = "^2023.7.22"
-aiohttp = "*"
-requests = "^2.31.0"
 
-uvicorn = {version = "^0.25.0", optional = true}
-fastapi = {version = "^0.104.1", optional = true}
-backoff = {version = "*", optional = true}
-pyyaml = {version = "^6.0", optional = true}
-rq = {version = "*", optional = true}
-orjson = {version = "^3.9.7", optional = true}
-streamlit = {version = "^1.29.0", optional = true}
-langchain = {version = "^0.1.0", optional = true}
-boto3 = {version = "^1.34.15", optional = true}
-google-generativeai = {version = "^0.3.2", optional = true}
-google-cloud-aiplatform = {version = "^1.39.0", optional = true}
-redis = {version = "^5.0.1", optional = true}
-anthropic = {version = "^0.9.0", optional = true}
-asyncio = {version = "^3.4.3", optional = true}
-cohere = {version = "^4.40", optional = true}
-langfuse = {version = "^2.5.0", optional = true}
-prisma = {version = "^0.11.0", optional = true}
-tenacity = {version = "^8.2.3", optional = true}
-traceloop-sdk = {version = "0.7.0", optional = true}
+python-dotenv ="^0.12.0"                                    # for env 
+tiktoken = "^0.4.0"                                         # for calculating usage
+importlib-metadata = "^6.8.0"                               # for random utils
+tokenizers = "^0.14.0"                                      # for calculating usage
+click = "^8.1.7"                                            # for proxy cli 
+jinja2 = "^3.1.2"                                           # for prompt templates
+certifi = "^2023.7.22"                                      # [TODO] clean up 
+aiohttp = "^3.9.0"                                          # for network calls
 
-# Missing due to conflict:
-# - baseten => depends on pytz (>=2022.7.1,<2023.0.0)
+anyio = {version = "^4.2.0", optional = true}               # openai + http req.
+openai = {version = "^1.0.0", optional = true}              # openai req. 
+fastapi = {version = "^0.100.0", optional = true}           # server dep
+pydantic = {version = "^2.5", optional = true}              # openai req. 
+backoff = {version = "^2.2.1", optional = true}             # server dep
+pyyaml = {version = "^6.0", optional = true}                # server dep
+uvicorn = {version = "^0.22.0", optional = true}            # server dep
+boto3 = {version = "^1.28.58", optional = true}             # aws bedrock/sagemaker calls
+redis = {version = "^4.6.0", optional = true}               # caching
+prisma = {version = "^0.11.0", optional = true}             # for db
+mangum = {version = "^0.17.0", optional = true}             # for aws lambda functions
+google-generativeai = {version = "^0.1.0", optional = true} # for vertex ai calls
+async_generator = {version = "^1.10.0", optional = true}    # for async ollama calls
```

Comment:
async_generator is no longer used @bufferoverflow, we moved our calls to httpx

Let me know where you saw this, will clean it up